### ISSUE TITLE
[BUG] `Groupby` requires groupby_cols in column selector

### ISSUE THREAD

**Issue by radekosmulski** [NVIDIA-Merlin/NVTabular#1599]
**Describe the bug**

`Grouby` requires `gropuby_cols` to be present in column selector. Reading the tests, my understanding is that if you don't include `groupby_cols` in column selector the op should run and the `groupby_cols` will not be included in the output?

Having `gropuby_cols` in output is problematic. You might start having numerical columns mixed with ones indicating a category and that renders many downstream nvt operations that you might want to perform on the output of `Groupby` impossible.

I created a PR that demonstrates this issue in a test and maybe can be helpful for working on this. #1598 

**Steps/Code to reproduce bug**
![image](https://user-images.githubusercontent.com/2444926/175697712-d93fa4cf-df3a-41c8-a765-7d76f3b0dac9.png)

```
import nvtabular as nvt
import cudf
import numpy as np

df = cudf.DataFrame(data={
    'product_id': list('abbcc'),
    'sales': np.random.randint(0,10,5),
})

out = ['product_id', 'sales'] >> nvt.ops.Groupby(groupby_cols=['product_id'], aggs='sum')


ds = nvt.Dataset(df)
wf = nvt.Workflow(out)

o = wf.fit_transform(ds).compute()
o
```

**Expected behavior**
Running Groupby without `groupby_cols` in column selector doesn't include `gropuby_cols` in output and the operation runs as intended.

**Environment details (please complete the following information):**
`nvcr.io/nvidia/merlin/merlin-tensorflow-training:22.05`, current main

**Comment by radekosmulski** [comment]
Just to provide a bit more background on the above, the test that I added currently fails, but I believe it should run, that would be the desired outcome.

We even have a test for this scenario that tests that this works on the schema side of things, but doesn't work when actual calculations are run.

```
def test_groupby_selector_cols():
    input_schema = Schema(["name", "id", "ts", "x", "y"])

    # Include the groupby col in the selector
    groupby = ColumnSelector(["name", "id", "ts", "x", "y"]) >> ops.Groupby(
        groupby_cols=["name"],
        sort_cols=["ts"],
        aggs={
            "x": ["list", "sum"],
            "y": ["first", "last"],
            "ts": ["min"],
        },
        name_sep="-",
    )
    workflow = Workflow(groupby).fit_schema(input_schema)

    # If groupby_cols are included in the selector, they should be in the output
    assert "name" in workflow.output_node.output_schema.column_names

    # Don't include the groupby col in the selector
    groupby = ColumnSelector(["id", "ts", "x", "y"]) >> ops.Groupby(
        groupby_cols=["name"],
        sort_cols=["ts"],
        aggs={
            "x": ["list", "sum"],
            "y": ["first", "last"],
            "ts": ["min"],
        },
        name_sep="-",
    )
    workflow = Workflow(groupby).fit_schema(input_schema)

    # If groupby_cols aren't included in the selector, they shouldn't be in the output
    assert "name" not in workflow.output_node.output_schema.column_names
```

The reason this is problematic is that because of `groupby_cols` being included in the output, there is a lot of ops we might want to run on the output of `Groupby` that we are unable to. The big case here is feeding data to a NN (any of the Merlin Models). In order for the NN to function well, we would need to first normalize the input data. But looking at even the example I included above, we cannot do this, because the workflow would try to attempt to normalize `product_id` as well and would fail.

![image](https://user-images.githubusercontent.com/2444926/175889024-542606ec-3d64-4c66-a224-ea592d8ef10d.png)

**Comment by viswa-nvidia** [comment]
@radekosmulski , please triage your bug with severity and priority
